src/components/Charts/ChataChart/ChataChart.js

```diff
@@ -0,0 +1,531 @@
+        const { deltaX, deltaY } = this.getDeltas()
+        const { innerHeight, innerWidth } = this.getInnerDimensions()
+        this.setState({ deltaX, deltaY, innerHeight, innerWidth }, () => {
+          this.adjustVerticalPosition()
+        })
+      }, 0)
+    }
+  }
+
+  getDeltas = () => {
+    if (this.props.type === 'pie') {
+      return { deltaX: 0, deltaY: 0 }
+    }
+
+    const axesBBox = getBBoxFromRef(this.innerChartRef?.chartRef)
+
+    // Get distance in px to shift to the right
+    const axesBBoxX = Math.ceil(axesBBox?.x ?? 0)
+    const deltaX = -1 * axesBBoxX + this.PADDING
+
+    // Get distance in px to shift down
+    const axesBBoxY = Math.ceil(axesBBox?.y ?? 0)
+    const deltaY = -1 * axesBBoxY + this.PADDING
+
+    return { deltaX, deltaY }
+  }
+
+  getRenderedChartDimensions = () => {
+    const leftAxisBBox = this.innerChartRef?.axesRef?.leftAxis?.ref?.getBoundingClientRect()
+    const bottomAxisBBox = this.innerChartRef?.axesRef?.bottomAxis?.ref?.getBoundingClientRect()
+    const rightAxisBBox = this.innerChartRef?.axesRef?.rightAxis?.ref?.getBoundingClientRect()
+    const topAxisBBox = this.innerChartRef?.axesRef?.topAxis?.ref?.getBoundingClientRect()
+    const clippedLegendBBox = this.innerChartRef?.axesRef?.legendRef?.legendClippingContainer?.getBoundingClientRect()
+    const axesBBox = mergeBboxes([leftAxisBBox, bottomAxisBBox, rightAxisBBox, topAxisBBox, clippedLegendBBox])
+
+    const axesWidth = axesBBox?.width ?? 0
+    const axesHeight = axesBBox?.height ?? 0
+    const axesX = axesBBox?.x ?? 0
+    const axesY = axesBBox?.y ?? 0
+
+    return {
+      chartHeight: axesHeight,
+      chartWidth: axesWidth,
+      chartX: axesX,
+      chartY: axesY,
+    }
+  }
+
+  getInnerDimensions = () => {
+    const { chartWidth, chartHeight } = this.getRenderedChartDimensions()
+
+    const propsWidth = typeof this.props.width === 'number' ? this.props.width : undefined
+    const propsHeight = typeof this.props.height === 'number' ? this.props.height : undefined
+
+    const containerWidth = propsWidth ?? this.chartContainerRef?.clientWidth ?? 0
+    const containerHeight = propsHeight ?? this.chartContainerRef?.clientHeight ?? 0
+
+    let innerWidth = containerWidth - 2 * this.PADDING
+    if (this.innerChartRef?.xScale && chartWidth) {
+      const rangeInPx = this.innerChartRef.xScale.range()[1] - this.innerChartRef.xScale.range()[0]
+      const totalHorizontalMargins = chartWidth - rangeInPx
+      innerWidth = containerWidth - totalHorizontalMargins - 2 * this.PADDING
+    }
+
+    let innerHeight = containerHeight - 2 * this.PADDING
+    if (this.innerChartRef?.yScale && chartHeight) {
+      const rangeInPx = this.innerChartRef.yScale.range()[0] - this.innerChartRef.yScale.range()[1]
+      const totalVerticalMargins = chartHeight - rangeInPx
+      innerHeight = containerHeight - totalVerticalMargins - 2 * this.PADDING
+    }
+
+    if (innerWidth < 1) {
```

Comment:
Could you Ceiling this value when it is calculated above? Or should some values have decimals?